### PR TITLE
[DBeX-60984] wrap form 0781* 3rd party action in logging

### PR DESCRIPTION
**Pull request by SamStuckey**
## Summary

- Wrap third party uploads of 0781 and 0781a forms in logging
- minor refactors to facilitate goal of "log as close to the transaction as possible" from the Epic.
- minor syntactic refactors to logging service

## Related issue(s)
- [ticket](https://app.zenhub.com/workspaces/disability-benefits-experience-team-carbs-6470c8bfffee9809b2634a52/issues/gh/department-of-veterans-affairs/va.gov-team/60984)
- [epic](https://app.zenhub.com/workspaces/disability-benefits-experience-team-carbs-6470c8bfffee9809b2634a52/issues/gh/department-of-veterans-affairs/va.gov-team/60952)

## Testing done

- Rspec tests cover existing functionality
- No user facing changes to verify
- changes are evident in logs when running rspec tests

## What areas of the site does it impact?
Form 526 Submission

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (N/A)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature (N/A)

### CODE REVIEW

File: app/workers/evss/disability_compensation_form/submit_form0781.rb [hunk 1]
```diff
@@ -2,12 +2,27 @@
...
+      attr_reader :submission_id, :evss_claim_id
+
+      wrap_with_logging(
+        :upload_to_vbms,
+        :perform_client_upload,
+        additional_instance_logs: {
+          submission_id: [:submission_id],
+          evss_claim_id: [:evss_claim_id]
+        }
+      )
```
Comment (SamStuckey) [inline app/workers/evss/disability_compensation_form/submit_form0781.rb:21]: Satisfy ticket requirement to wrap third party interactions in logging.

File: app/workers/evss/disability_compensation_form/submit_form0781.rb [hunk 2]
```diff
@@ -42,38 +57,41 @@ class SubmitForm0781 < Job
       # `file`, which is the generated file location
       def get_docs(submission_id, uuid)
         @submission_id = submission_id
-        parsed_forms = JSON.parse(submission.form_to_json(Form526Submission::FORM_0781))
-        file_type_and_file_objs = []
+        @uuid = uuid
```
Comment (SamStuckey) [inline app/workers/evss/disability_compensation_form/submit_form0781.rb:60]: making this an instance variable makes the value available to the instance level logging.

File: app/workers/evss/disability_compensation_form/submit_form0781.rb [hunk 3]
```diff
@@ -42,38 +57,41 @@ class SubmitForm0781 < Job
...
+
         { 'form0781' => FORM_ID_0781, 'form0781a' => FORM_ID_0781A }.each do |form_type_key, actual_form_types|
           if parsed_forms[form_type_key].present?
             file_type_and_file_objs << {
               type: actual_form_types,
-              file: process_0781(submission.auth_headers, uuid, FORM_ID_0781, parsed_forms[form_type_key],
+              file: process_0781(uuid, FORM_ID_0781, parsed_forms[form_type_key],
```
Comment (SamStuckey) [inline app/workers/evss/disability_compensation_form/submit_form0781.rb:66]: `submission` is globally available, so no need to pass it as an argument.

File: app/workers/evss/disability_compensation_form/submit_form0781.rb [hunk 4]
```diff
@@ -42,38 +57,41 @@ class SubmitForm0781 < Job
...
                                  upload: false)
             }
           end
         end
         file_type_and_file_objs
       end
 
+      def parsed_forms
```
Comment (SamStuckey) [inline app/workers/evss/disability_compensation_form/submit_form0781.rb:74]: minor refactor to de-dup this shared functionality.

File: app/workers/evss/disability_compensation_form/submit_form0781.rb [hunk 5]
```diff
@@ -42,38 +57,41 @@ class SubmitForm0781 < Job
...
+        @parsed_forms ||= JSON.parse(submission.form_to_json(Form526Submission::FORM_0781))
+      end
+
       # Performs an asynchronous job for generating and submitting 0781 + 0781A PDF documents to VBMS
       #
       # @param submission_id [Integer] The {Form526Submission} id
       #
       def perform(submission_id)
+        @submission_id = submission_id
```
Comment (SamStuckey) [inline app/workers/evss/disability_compensation_form/submit_form0781.rb:83]: assigning this to an attribute / instance var makes it available to the instance level logger

File: app/workers/evss/disability_compensation_form/submit_form0781.rb [hunk 6]
```diff
@@ -124,23 +143,31 @@ def create_document_data(evss_claim_id, upload_data)
...
       ensure
         # Delete the temporary PDF file
         File.delete(pdf_path) if pdf_path.present?
       end
+
+      def perform_client_upload(file_body, document_data)
+        client.upload(file_body, document_data)
+      end
+
+      def client
```
Comment (SamStuckey) [inline app/workers/evss/disability_compensation_form/submit_form0781.rb:164]: this minor refactor is more readable, and also makes client available in our `perform_client_upload` method without needing to pass it as an argument

File: lib/logging/third_party_transaction.rb [hunk 7]
```diff
@@ -27,10 +27,11 @@ def wrap_with_logging(*method_names, additional_class_logs: {}, additional_insta
           method_names.each do |method_name|
             # define patchable method(s) with the same name inside of a proxy module.
             define_method(method_name) do |*args, &block|
-              log_3pi_begin(method_name, additional_class_logs, additional_instance_logs, *args)
+              str_args = args.to_s
+              log_3pi_begin(method_name, additional_class_logs, additional_instance_logs, str_args)
```
Comment (SamStuckey) [inline lib/logging/third_party_transaction.rb:31]: no need to pass full arguments when a string will do.  All we are doing is adding this stringified arg chain to our log.

File: lib/logging/third_party_transaction.rb [hunk 8]
```diff
@@ -44,7 +45,7 @@ def wrap_with_logging(*method_names, additional_class_logs: {}, additional_insta
     # these will be included after instance instantiation, making them available
     # to the instance and retaining their scope.
     module ScopedInstanceMethods
-      def log_3pi_begin(method_name, additional_class_logs = {}, additional_instance_logs = {}, args = [])
```
Comment (SamStuckey) [inline lib/logging/third_party_transaction.rb:47]: These default args are redundant, as the calling method already has defaults set.

**Review by aurora-a-k-a-lightning — COMMENTED** [review]
lgtm! :tada:

**Review by ericboehs — APPROVED** [review]

**Pull request merged**